File: webapp/src/app/game-summary.ts
import {Color} from './color.enum'
import {GameState} from './game-state.enum'
import {User} from './user'

/**
 * Game summary.
 */
export interface GameSummary {
  /**
   * ID.
   */
  id: string

  /**
   * Creation date (ISO 8601).
   */
  creationDate: string

  /**
   * White player.
   */
  whitePlayer: User

  /**
   * Black player.
   */
  blackPlayer: User

  /**
   * Color for which is turn to play.
   */
  turnColor: Color

  /**
   * State.
   */
  state: GameState

  /**
   * Color of winner or null if the game is not checkmate.
   */
  winnerColor: Color | undefined
}

File: api/src/main/kotlin/dev/gleroy/ivanachess/api/game/GameSummary.kt
package dev.gleroy.ivanachess.api.game

import dev.gleroy.ivanachess.api.Entity
import dev.gleroy.ivanachess.api.user.User
import dev.gleroy.ivanachess.core.Game
import dev.gleroy.ivanachess.core.Piece
import java.time.OffsetDateTime
import java.util.*

/**
 * Game summary.
 *
 * @param id ID.
 * @param creationDate Creation date.
 * @param whitePlayer User who plays white color.
 * @param blackPlayer User who plays black color.
 * @param turnColor Color for which is turn to play.
 * @param state Game state.
 */
data class GameSummary(
    override val id: UUID = UUID.randomUUID(),
    override val creationDate: OffsetDateTime = OffsetDateTime.now(),
    val whitePlayer: User,
    val blackPlayer: User,
    val turnColor: Piece.Color = Piece.Color.White,
    val state: Game.State = Game.State.InGame
) : Entity
